webrecorder/webrecorder/rec/webrecrecorder.py

```diff
@@ -560,20 +557,31 @@ def add_urls_to_index(self, stream, params, filename, length):
         cdx_list = (super(WebRecRedisIndexer, self).
                       add_urls_to_index(stream, params, filename, length))
 
+
+        # if replay key exists, add to it as well!
```

Comment:
Well deserved exclamation point. Being able to continuously update a cache that you can replay at turbo speed is the holy grail.

I've had some luck getting PyWB to work as a transparent forward proxy using some of the Redis indexing and replay improvements you've been shifting over from this repo, but only by doing bulk loads of data using CDX queries and then replaying from local archives.

Webrecorder is clearly better suited to what I'm looking for (time traveling web crawling). That NGINX forward proxying setup is super clever, and makes the whole project a far more polite internet citizen. I just haven't been able to figure out where is the easiest place to hook in without taking the whole user administration, email marketing, public website, etc parts along for the ride.

The full stack runner is the shortest path, but I feel like there is some magical combination of config flags that get me a workable proxy in a more scalable way closer to the main code base. Is there a config file somewhere that takes advantage of this extract/patch/replay workflow in something closer to a headless proxy environment. I can see all the pieces I need are there, but I'm just not sure where to start putting them together.

Reply:
Webrecorder actually already does this, keeping everything in Redis. This PR flushes the CDXJ to disk, and reloads them as needed, and also consolidates replay to a single key per collection.

That being said, a lot of the core functionality is in pywb, though more work and documentation is happening.. We are also trying to make Webrecorder more modular so you can deploy it on your own as well.. Can you provide more info about your use case and we can offer more suggestions/help support it. Please send us an e-mail us at support@webrecorder.io